qiskit/user_config.py

```diff
@@ -146,6 +146,39 @@ def read_config_file(self):
                 self.settings['num_processes'] = num_processes
 
 
+def set_config(key, value, section=None):
+    """Adds or modifies a user configuration
+
+    It will add the configuration in either the default location
+    ~/.qiskit/settings.conf or if set the value of the `QISKIT_SETTINGS` env var.
+
+    Args:
+        key (str): name of the config
+        value (str): value of the config
+        section (str, optional): if not specified, adds it to the
+            `default` section of the config file.
+    """
+    file_name = os.getenv('QISKIT_SETTINGS', DEFAULT_FILENAME)
+    sec_name = section if section is not None else 'default'
+
+    assert isinstance(key, str), 'Key must be string type'
+    assert isinstance(value, str), 'Value must be string type'
```

Comment:
In general, the style we are following is to raise our own exceptions instead of `AssertionError`. For this particular case, it looks like this:
```suggestion
    if not isinstance(key, str):
        raise exceptions.QiskitUserConfigError('Key must be string type')
    if not isinstance(value, str):
        raise exceptions.QiskitUserConfigError('Value must be string type')
```